esp32/boards: Add Silicognition ManT1S.

xorbit opened the pull request:
<!-- Thanks for submitting a Pull Request! We appreciate you spending the
     time to improve MicroPython. Please provide enough information so that
     others can review your Pull Request.

     Before submitting, please read:
     https://github.com/micropython/micropython/blob/master/CODEOFCONDUCT.md
     https://github.com/micropython/micropython/wiki/ContributorGuidelines

     Please check any CI failures that appear after your Pull Request is opened.
-->

### Summary

<!-- Explain the reason for making this change. What problem does the pull request
     solve, or what improvement does it add? Add links if relevant. -->

Add new board planned to be launched on Crowd Supply September 2025.
The ManT1S is an ESP32-based core module with IEEE 802.3cg 10BASE-T1S Single Pair Ethernet networking and power distribution over data lines.

Pre-launch page is not up yet unfortunately, or I'd add a link to it.  Any day now hopefully, I can add a comment here later if needed.  I plan to ship these with MicroPython pre-loaded.  I'm trying to get a head start with getting this merged before the next release since I prefer to ship with official MicroPython releases.

### Testing

<!-- Explain what testing you did, and on which boards/ports. If there are
     boards or ports that you couldn't test, please mention this here as well.

     If you leave this empty then your Pull Request may be closed. -->

Tested on prototypes of my new ManT1S boards, most recent tests on latest revision 4 hardware.
I've been testing LAN8670 integration on MicroPython from before this was officially supported, see my comments in #15731 and #16421.

xorbit committed:
esp32/boards: Add Silicognition ManT1S.

New board planned to be launched on Crowd Supply September 2025.
ESP32-based core module with IEEE 802.3cg 10BASE-T1S Single Pair
Ethernet networking and power distribution over data lines.

Signed-off-by: Patrick Van Oosterwijck <patrick@silicognition.com>

xorbit committed:
esp32/boards: Fix formatting for Silicognition ManT1S.

Signed-off-by: Patrick Van Oosterwijck <patrick@silicognition.com>

dpgeorge commented on ports/esp32/boards/SIL_MANT1S/board.json:
```diff
@@ -0,0 +1,26 @@
+        "BLE",
+        "External Flash",
+        "External RAM",
+        "WiFi",
```
Remove the comma at the end of this line.

dpgeorge commented on ports/esp32/boards/SIL_MANT1S/partitions-8MiB-ota.csv:
```diff
@@ -0,0 +1,10 @@
+# Partition table for MicroPython with OTA support using 8MB flash
```
Do you need this custom partition table?  Can you use the existing `partitions-8MiBplus-ota.csv` instead?

If you need a custom one, please remove the `vfs` entry.  It's now automatically detected, along with the size of the flash (so the firmware will work with any board that has at least 8MiB flash).

xorbit replied:
I looked at the standard partition table and it seems to be great for 16 MiB flash (seems to match what I had for the wESP32) but with 8 MiB it seems to leave very little room for the vfs.

The standard one reserves 2.43 MiB for each OTA partition leaving only 3 MiB for the vfs.  In most cases this is probably fine, but this thing has a good chance of having a web server running on it with graphics that may fill it up fast.  Also I have to consider that to make OTA work, I need to reserve enough space to store an OTA image on the vfs before flashing it.  With 2.43MiB max image size, the user would have to store maximum 0.5 MiB on the vfs which is quite limiting.

With my custom partition table I have 2.0 MiB for each OTA partition, leaving just under 4 MiB for the vfs.  If the user needs to reserve 2 MiB for a possible OTA image, their application can use up to 2 MiB which seems much more reasonable.

The ManT1S image currently builds to about 1.5 MiB, which leaves another 0.5 MiB for it to grow in the future.  Do you have reason to believe this will soon be insufficient?

xorbit committed:
esp32/boards: Fix Silicognition ManT1S config files.

Remove errant comma in board.json.
Remove vfs line from partitions-8MiB-ota.csv.

Signed-off-by: Patrick Van Oosterwijck <patrick@silicognition.com>

xorbit commented:
Shoot, just a few days too late.  What's your release cadence again?  I don't expect to ship before the end of the year.

dpgeorge commented:
> What's your release cadence again? I don't expect to ship before the end of the year.

About 3 months, so the next release will be before the end of this year.

xorbit commented:
Excellent, that should do fine!

xorbit commented:
Hmm, I was trying to get this up to date with master, but it looks like git has done something weird and made all those changes in master part of my PR.  Have I made it worse or is this OK?

dpgeorge commented:
> Hmm, I was trying to get this up to date with master, but it looks like git has done something weird and made all those changes in master part of my PR. Have I made it worse or is this OK?

You made it worse :sweat_smile: 

You'll need to rebase on latest master and force push.  I also suggest you use a branch from your own repo/fork, instead of using your master.

xorbit committed:
esp32/boards: Add 10BASE-T1S to non filterable features.

Signed-off-by: Patrick Van Oosterwijck <patrick@silicognition.com>

xorbit commented:
Ok I got the rebase done.  I don't know how to switch this PR to a different branch so I left it on my master.

robert-hh commented:
>  I don't know how to switch this PR to a different branch

Just `git checkout -b branch_name`
or 
`git branch branch_name`

That will create the branch `branch_name` with the initial state of your actual master. The first variant will include the checkout of the new branch.

xorbit commented:
Yes I do know that but I don't know how to switch this PR to the branch.
When I click "Edit" on top of the PR it does not let me change the "from [xorbit:master]" to something else.

I am not using my fork for anything other than getting this PR merged so I'd like to see this merged without requiring any more work on my part.  It should be possible to merge this from my master just fine.

robert-hh commented:
AFAIK it is not possible to change the source branch of a commit. One has to create a new PR. In that case, just leave it at `master`.

xorbit commented:
I wanted to mention that I tested this to work with `espressif/lan867x` component versions 1.0.3 and 2.0.0.  Version 1.0.x is what's currently requested in `idf_component.yml`, but I don't consider updating the component version part of this PR. This PR is just for adding a board and it works with either version.

It might be good to push the component version to 2.0.0 though since it changes the interface to add PLCA support which I might want to implement at some point.
The original `espressif/lan867x` component version 1.0.x was added in #16421.

dpgeorge commented on ports/esp32/boards/SIL_MANT1S/board.json:
```diff
@@ -0,0 +1,27 @@
+    "mcu": "esp32",
+    "product": "ManT1S",
+    "thumbnail": "",
+    "url": "https://mant1s.net/",
```
This URL doesn't seem to work.  What is its status?

xorbit replied:
Hadn't gotten around to it yet.  Placeholder start-of-a-site is live now.

dpgeorge commented on ports/esp32/boards/SIL_MANT1S/mpconfigboard.cmake:
```diff
@@ -0,0 +1,10 @@
+set(SDKCONFIG_DEFAULTS
+    boards/sdkconfig.base
+    ${SDKCONFIG_IDF_VERSION_SPECIFIC}
```
This line is no longer needed.

xorbit replied:
Removed it.

xorbit committed:
esp32/boards: Remove unnecessary SDKCONFIG_IDF_VERSION_SPECIFIC.

mpconfigboard.cmake doesn't require
`${SDKCONFIG_IDF_VERSION_SPECIFIC}` line any more.

Signed-off-by: Patrick Van Oosterwijck <patrick@silicognition.com>

xorbit commented:
Am I missing the merge window for the next version AGAIN?
Despite this dragging on and being ready for months?  For what reason?
I do not understand why this has to be so difficult.  It's a simple new board merge. :(

dpgeorge commented:
> Despite this dragging on and being ready for months? For what reason?

Ideally we would not merge a board definition for hardware that doesn't yet exist, because in can't (yet) be tested or used by others.  And if the board does not launch, then the definition would need to be deleted (that's not a big deal, but adds a burden and code churn).

But it sounds like you're committed to bring it to life, so we can make an exception here.

(Also, would be nice if the `ping.py` code was merged to `micropython-lib` and then referenced in the `manifest.py`, instead of adding it here verbatim.  That's not a show stopper though.)

xorbit commented:
The parts are ordered, so yes it will be brought to life.
